java/adapter/avro/src/main/java/org/apache/arrow/consumers/Consumer.java

```diff
@@ -33,8 +33,12 @@
    */
   void consume(Decoder decoder) throws IOException;
 
+  void consume(Decoder decoder, int index) throws IOException;
```

Comment:
docs.  Also is there a reason that both overloads are necessary?  If they really are it would be nice to have one implementation delegate to different one.

Reply:
This is mainly for union type, when write data to its subVector, the index for subVector is not   continuous, so when union consumer consume a value via delegate, the index should be specified.
You are right, two overloads seems ugly and not necessary, I add setPosition(int index) API to set index instead.